File: ProjectEuler/003/003.py
import math

def smallest_factor(n):
	if(n <= 1): raise Exception("1")
	for i in range(2, math.ceil(math.sqrt(n))):
		if(n % i == 0): return i;
	return n;

n = 600851475143
p = smallest_factor(n);
while p < n :
	n /= p;
	p = smallest_factor(n);
print(n);

File: ProjectEuler/003/003.cpp
#include <iostream>
#include <cmath>

long smallest_factor(long n){
	if(n <= 1) throw 20;
	for(long i = 2; i < ceil(sqrt(n)); i++){
		if(n % i == 0) return i;
	}
	return n;
}

int main(){
	long n = 600851475143;
	long p = smallest_factor(n);
	while(p < n){
		n /= p;
		p = smallest_factor(n);
	}
	std::cout << n;

	return 0;
}
